### PR TITLE
[tests-only][full-ci]setup app provider in `CI`

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
This PR make the test run in CI  merged at https://github.com/owncloud/web/pull/9265 for app-provider office suites.

### Related Issue:
https://github.com/owncloud/web/issues/9191

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -741,12 +754,20 @@ def e2eTests(ctx):
...
         depends_on = ["cache-ocis"]
 
-        # oCIS specific steps
-        steps += setupServerConfigureWeb(params["logLevel"]) + \
-                 restoreOcisCache() + \
-                 (tikaService() if params["tikaNeeded"] else []) + \
-                 ocisService("e2e-tests", tika_enabled = params["tikaNeeded"]) + \
-                 getSkeletonFiles()
+        if suite == "oCIS-app-provider":
+            # oCIS tests app-provider to office suites
```
Comment (saw-jan) [inline .drone.star:758]:
```suggestion
            # app-provider specific steps
```

File: .drone.star [hunk 2]
```diff
@@ -1237,18 +1258,22 @@ def ocisService(type, tika_enabled = False):
...
         "WEB_ASSET_PATH": "%s/dist" % dir["web"],
-        "WEB_UI_CONFIG": "%s" % dir["ocisConfig"],
         "FRONTEND_SEARCH_MIN_LENGTH": "2",
         "FRONTEND_OCS_ENABLE_DENIALS": True,
     }
+    if type == "app-provider":
+        environment["GATEWAY_GRPC_ADDR"] = "0.0.0.0:9142"
+        environment["MICRO_REGISTRY"] = "mdns"
+    else:
+        environment["WEB_UI_CONFIG"] = "%s" % dir["ocisConfig"]
```
Comment (saw-jan) [inline .drone.star:1269]: will web_config also affect app-provider?

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 3]
```diff
@@ -336,6 +346,24 @@ export const openAndGetContentOfDocument = async ({
...
+const isAppProviderServiceForOfficeSuitesReadyInWebUI = async (page, type) => {
+  let retry = 1
+  let createNewOfficeDocumentFileButtonLocator
+  while (retry <= 5) {
+    createNewOfficeDocumentFileButtonLocator = await page
+      .locator(util.format(createNewOfficeDocumentFileBUtton, type))
+      .isVisible()
+    if (createNewOfficeDocumentFileButtonLocator === true) {
+      return true
+    }
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:358]: let's say it break
```suggestion
    if (createNewOfficeDocumentFileButtonLocator === true) {
      break
    }
```

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 4]
```diff
@@ -336,6 +346,24 @@ export const openAndGetContentOfDocument = async ({
...
+    await new Promise((resolve) => setTimeout(resolve, 3000))
+    await page.reload()
+    await page.locator(addNewResourceButton).click()
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:361]:
```suggestion
```

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 5]
```diff
@@ -336,6 +346,24 @@ export const openAndGetContentOfDocument = async ({
   return actualContentOfEditor
 }
 
+const isAppProviderServiceForOfficeSuitesReadyInWebUI = async (page, type) => {
+  let retry = 1
+  let createNewOfficeDocumentFileButtonLocator
+  while (retry <= 5) {
+    createNewOfficeDocumentFileButtonLocator = await page
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:353]:
```suggestion
  while (retry <= 5) {
    await page.locator(addNewResourceButton).click()
    createNewOfficeDocumentFileButtonLocator = await page
```

**Review by saw-jan — COMMENTED** [review]
open dropdown and then check?

File: .drone.star (hunk 2 above)
Comment (SagarGi) [reply .drone.star]: no it did not work https://drone.owncloud.com/owncloud/web/38160/12/18

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 6]
```diff
@@ -267,6 +267,18 @@ const createDocumentFile = async (
...
+  // for creating office suites documents we need the external app provider services to be ready
+  // though the service is ready it takes some time for the list of office suites documents to be visible in the dropdown in the webUI
+  // which requires a retry to check if the service is ready and the office suites documents is visible in the dropdown
+  const isAppProviderServiceReadyInWebUI = await isAppProviderServiceForOfficeSuitesReadyInWebUI(
+    page,
+    type
+  )
+  if (isAppProviderServiceReadyInWebUI === false) {
+    throw new Error(
+      `The document of type ${type} did not appeared in the webUI for ${editorToOpen}. Possible reason could be the app provider service for ${editorToOpen} was not ready yet.`
```
Comment (phil-davis) [inline tests/e2e/support/objects/app-files/resource/actions.ts:279]:
```suggestion
      `The document of type ${type} did not appear in the webUI for ${editorToOpen}. Possible reason could be the app provider service for ${editorToOpen} was not ready yet.`
```

**Review by phil-davis — APPROVED** [review]
LGTM. I only noticed one tiny grammar thing in a message.

**Review by saw-jan — APPROVED** [review]

**Pull request merged**